### PR TITLE
[13.0] Add module partner_tz

### PR DESCRIPTION
**Pull request by grindtildeath**
This module removes timezone default value on res.partner and display the field
on form view.

### CODE REVIEW

**Review by jgrandguillaume — APPROVED** [review]
LGTM as far as I can tell

File: setup/partner_tz/odoo/addons/partner_tz [hunk 1]
```diff
@@ -0,0 +1 @@
+../../../../partner_tz
```
Comment (Tonow-c2c) [inline setup/partner_tz/odoo/addons/partner_tz:1]: this work in every project scaffold ?

File: partner_tz/tools/tz_utils.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    :return: tz-naive datetime.datetime object
+    """
+    if isinstance(from_tz, str):
+        from_tz = pytz.timezone(from_tz)
+    if isinstance(to_tz, str):
+        to_tz = pytz.timezone(to_tz)
+    return from_tz.localize(date_time).astimezone(to_tz).replace(tzinfo=None)
+
+
+def tz_to_utc_naive_datetime(from_tz, date_time):
```
Comment (Tonow-c2c) [inline partner_tz/tools/tz_utils.py:26]: it's odoo core method and you override them ?

File: partner_tz/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+{
+    "name": "Partner timezone",
+    "summary": "Remove partner timezone default value and display on form",
+    "version": "13.0.1.0.0",
+    "development_status": "Alpha",
+    "category": "Uncategorized",
```
Comment (Tonow-c2c) [inline partner_tz/__manifest__.py:8]: May be the category can be improve

File: setup/partner_tz/odoo/addons/partner_tz (hunk 1 above)
Comment (grindtildeath) [reply setup/partner_tz/odoo/addons/partner_tz:1]: IIRC these setup files are regenerated after merges by the OCA git bot...

File: partner_tz/tools/tz_utils.py (hunk 2 above)
Comment (grindtildeath) [reply partner_tz/tools/tz_utils.py:26]: It is not, it's new helpers tools if one must do tz computations.

File: partner_tz/__manifest__.py (hunk 3 above)
Comment (grindtildeath) [reply partner_tz/__manifest__.py:8]: Do you have a suggestion? It's only depending on base, and looking at other modules in this repo, I'm not sure there are much better fits :shrug:

**Review by LoisRForgeFlow — COMMENTED** [review]
Looks good, could you check travis error?

**Comment by LoisRForgeFlow** [comment]
@grindtildeath Hi, this module is a dependency for several modules you are proposing, if we get this merged it will ease testing and attract reviewers. Can you take a few minutes to fix travis and push this forward? :rocket:

File: partner_tz/tools/tz_utils.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
+# Copyright 2020 Camptocamp SA
```
Comment (NL66278) [inline partner_tz/tools/tz_utils.py:1]: The functions presented here are very thin wrappers around the original functions from pytz. I think this is of questionable value. Better use the original pytz functions that are generally available and not depending on having this particular module installed.

**Review by NL66278 — CHANGES_REQUESTED** [review]
Please limit the module to adding the timezone to the partner form.

But you might think about the suggested functionality of setting the default partner timezone based on the partners country...

**Review by HaraldPanten — APPROVED** [review]
Functional review

File: partner_tz/tools/tz_utils.py (hunk 4 above)
Comment (grindtildeath) [reply partner_tz/tools/tz_utils.py:1]: Thx for your review @NL66278 
The value of a function is quite subjective in any case, but to me they make tz conversions more explicit and avoid repeats. Moreover they are only tools that won't change anything when someone installs this module.
So I'm sorry but I don't see why having this file here, and using it in other modules having this module as a dependency would be a problem...

**Comment by rousseldenis** [comment]
@grindtildeath I'd prefer explicit words in module name, partner_timezone is better

**Review by rousseldenis — APPROVED** [review]
Code review. Apart module name

**Comment by rousseldenis** [comment]
And some tests for conversion functions should be great

**Comment by jgrandguillaume** [comment]
Merging as Alpha version is set. Thank everyone for the review.

**Pull request merged**